db/transforms/operations/finish_specifying.py

```diff
@@ -0,0 +1,156 @@
+from db.tables.operations.select import get_joinable_tables
+from db.transforms.base import Summarize, SelectSubsetOfColumns
+from db.columns.operations.select import get_column_from_oid_and_attnum
+
+
+def finish_specifying_summarize_transform(db_query, ix_of_summarize_transform, engine, metadata):
+    """
+    Will find initial columns that are not mentioned in the summarize_transform and will add each
+    of them to its grouping set and/or aggregating set.
+
+    If the user selected initial column (summarize's base grouping column) is not unique
+    constrained, will put the unmentioned initial columns in the aggregation set.
+
+    If the user selected initial column (summarize's base grouping column) is unique, then it might
+    put the initial column in the grouping set, depending on what _should_group_by returns.
+    """
+    summarize_transform = db_query.transformations[ix_of_summarize_transform]
+    assert type(summarize_transform) is Summarize
+    initial_columns_not_in_summarize = _get_initial_columns_not_in_summarize(db_query, summarize_transform)
+    if not initial_columns_not_in_summarize:
+        # If all input aliases for this transform are mentioned in the group-agg transform spec, we
+        # consider it fully specified.
+        return summarize_transform
+    # A group-agg transform has a base_grouping_column, or a user-selected input alias around
+    # which our suggestions will be based.
+    user_selected_alias = summarize_transform.base_grouping_column
```

Comment:
`base_grouping_column` is clearer to me. Down lower in the code, there's no good way to know what `user_selected_alias` refers to. Further, we shouldn't know or care whether a user selected it, or whether it was selected algorithmically, etc. at this point. I'd be okay with `base_grouping_alias` if needed.

Reply:
Sure, `user_selected` is redundant once we have the `base_grouping` qualifier. I liked `user_selected` because it's easier to understand than `base_grouping`, but, as you note, it's not general enough.